mlflow/store/sqlalchemy_store.py

```diff
@@ -2,27 +2,92 @@
 import uuid
 
 from mlflow.entities.lifecycle_stage import LifecycleStage
+from mlflow.store.dbmodels.db_types import DB_ENGINES, MYSQL
 from mlflow.store.dbmodels.models import Base, SqlExperiment, SqlRun, SqlMetric, SqlParam, SqlTag
 from mlflow.entities import RunStatus, SourceType
 from mlflow.store.abstract_store import AbstractStore
 from mlflow.entities import ViewType
 from mlflow.exceptions import MlflowException
 from mlflow.protos.databricks_pb2 import INVALID_PARAMETER_VALUE, RESOURCE_ALREADY_EXISTS, \
     INVALID_STATE, RESOURCE_DOES_NOT_EXIST
+from mlflow.utils.file_utils import build_path, exists, mkdir
 from mlflow.utils.mlflow_tags import MLFLOW_PARENT_RUN_ID, MLFLOW_RUN_NAME
 from mlflow.utils.search_utils import does_run_match_clause
 
 
+DEFAULT_ARTIFACT_ROOT = "artifact_root"
+
+
+class DbHelper:
+    def __init__(self, db_type, session):
+        if db_type not in DB_ENGINES:
+            raise MlflowException("Unsupported DB engine '{}'. "
+                                  "Valid types are: {}'".format(db_type, DB_ENGINES))
+        self.db_type = db_type
+        self.session = session
+
+    def __enter__(self):
+        if self.db_type == MYSQL:
+            self.session.execute("SELECT @@GLOBAL.sql_mode;")
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        if self.db_type == MYSQL:
+            self.session.execute("SELECT @@GLOBAL.sql_mode='';")
+
+
 class SqlAlchemyStore(AbstractStore):
+    ARTIFACTS_FOLDER_NAME = "artifacts"
+
+    def __init__(self, db_uri, default_artifact_root=None):
 
-    def __init__(self, db_uri):
         super(SqlAlchemyStore, self).__init__()
+        self.db_uri = db_uri
+        self.artifact_root_uri = default_artifact_root or DEFAULT_ARTIFACT_ROOT
         self.engine = sqlalchemy.create_engine(db_uri)
         Base.metadata.create_all(self.engine)
         Base.metadata.bind = self.engine
         db_session = sqlalchemy.orm.sessionmaker(bind=self.engine)
         self.session = db_session()
 
+        if self.artifact_root_uri == DEFAULT_ARTIFACT_ROOT and not exists(DEFAULT_ARTIFACT_ROOT):
+            mkdir(DEFAULT_ARTIFACT_ROOT)
+
+        if len(self.list_experiments()) == 0:
+            self._create_default_experiment(db_uri.split(':')[0])
```

Comment:
For safety, could we use `urlparse(db_uri).scheme`? I don't know if there are any weird shenanigans that could happen, but URIs are complicated.